src/modules/expression/expr.rs

```diff
@@ -101,6 +101,14 @@ impl Typed for Expr {
 }
 
 impl Expr {
+    pub fn get_integer_value(&self) -> Option<isize> {
+        match &self.value {
+            Some(ExprType::Number(value)) => value.get_integer_value(),
+            Some(ExprType::Neg(value)) => value.get_integer_value(),
+            _ => None,
+        }
+    }
```

Comment:
This logic has to be here, because the `Some(ExprType::Neg(value))` match arm calls `get_integer_value()` on the `Neg` struct, which unboxes its encapsulated `Box<Expr>` and calls this function on the result.

Reply:
See my reply to your other comment below.